python/lib/scheduler/dmod/scheduler/job/job_manager.py

```diff
@@ -532,31 +529,37 @@ def delete_job(self, job_id) -> bool:
         ValueError
             If no job exists with given job id.
         """
-        logging.debug("Deleting job {}".format(job_id))
+        if not isinstance(max_attempt_count, int) or max_attempt_count < 1:
+            max_attempt_count = 5
+
+        logging.debug(f"Deleting job {job_id}")
+
         # Retrieve the job and ensure any allocations are release
         job_key = self._get_job_key_for_id(job_id)
         job_obj = self.retrieve_job_by_redis_key(job_key)
 
         # Ensure allocations are released
         self.release_allocations(job_id)
 
-        i = 0
-        while i < 5:
-            i += 1
+        most_recent_exception = None
+
+        for _ in range(max_attempt_count):
             with self.redis.pipeline() as pipeline:
                 if job_obj.status.is_active:
                     # Make sure not in active set
                     pipeline.srem(self._active_jobs_set_key, job_key)
                 pipeline.delete(job_key)
                 pipeline.execute()
+
                 # Try to do this, but don't fully fail just for this part
                 try:
                     job_obj.rsa_key_pair.delete_key_files()
-                except:
-                    # TODO: look at at least logging something if this happens; for now ...
-                    pass
-                return True
+                    return True
+                except BaseException as exception:
+                    most_recent_exception = exception                    
+        
         # If we get here, it means we failed the max allowed times, so bail
+        logging.error(f"The job with ID '{job_id}' could not be deleted", exc_info=most_recent_exception)
```

Comment:
Uhhh, you could have up to 5 errors here. Ideally we would report all of these exceptions as they could be unique, although not likely.

Reply:
The cleanest way to handle this would be with an `ExceptionGroup`, but that's a 3.11 feature (i.e. not in our minimum version). As I see it, my options here are maintaining the error from the latest attempt or combine formatted messages from each error. That can get hairy if/when tracebacks/frames or printing become involved. Something smart is always nice, but that's a little out of scope of "let's slap something on there for an X% improvement". Should I add a TODO or something?

Reply:
Its better than it was when we started, so I think it is fine to keep it as is.

Yeah, ive been temped in several places in the code base to bring in the [`exceptiongroup`](https://pypi.org/project/exceptiongroup/) back port of that feature. Its harder to handle without pattern matching though.